snooty/parser.py

```diff
@@ -573,9 +573,11 @@ def dispatch_departure(self, node: tinydocutils.nodes.Node) -> None:
             self.handle_method_option(popped)
 
         elif isinstance(popped, n.Directive) and popped.name == "collapsible":
-            section = popped.get_child_of_type(n.Section)
-            heading = popped.get_child_of_type(n.Heading)
-            if not section and not heading:
+            html5_id = util.make_html5_id(popped.options.get("heading", "")).lower()
+            popped.options["id"] = html5_id
+            has_section = any(True for _ in popped.get_child_of_type(n.Section))
+            has_heading = any(True for _ in popped.get_child_of_type(n.Heading))
+            if not has_section and not has_heading:
```

Comment:
Would it make sense to create a section regardless of the contents of the collapsible? Seems like we're always increasing the section depth in the `ContentsHandler` when a `collapsible` is entered anyways.

The [updated staging link](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/DOCSP-43269-2/cloud-docs/seung.park/DOP-4983-frontend/tutorial/insert-data-cli/#required-access) still shows an h2 nested under the last collapsible. My understanding is that "Required Access" should be an h3, similar to the headings of the steps in the second collapsible.

Reply:
I think two sections *should* be fine. The structure would be somewhat similar to if an h2 heading immediately wraps an h3 heading. The main difference is the order of the section (section -> h2 -> section -> h3 vs. collapsible -> section -> section -> h3). Wrapping the collapsible in a section like you suggested should help resolve the issue while keeping order consistent, so I think that's also a good idea

Reply:
I've also updated that above staging link to be parsed from this branch!